convert2rhel/actions/pre_ponr_changes/subscription.py

```diff
@@ -16,46 +16,97 @@
 __metaclass__ = type
 
 import logging
+import os.path
 
-from convert2rhel import actions, cert, pkghandler, repo, subscription, toolopts
+from convert2rhel import actions, backup, cert, pkghandler, repo, subscription, toolopts, utils
 
 
 logger = logging.getLogger(__name__)
 
+# Source and target directories for the cdn.redhat.com domain ssl ca cert that:
+# - we tell customers to use when installing convert2rhel from that domain
+# - is used by RHSM when accessing RHEL repos hosted on the Red Hat CDN
+_REDHAT_CDN_CACERT_SOURCE_DIR = utils.DATA_DIR
+_REDHAT_CDN_CACERT_TARGET_DIR = "/etc/rhsm/ca/"
+
+# Source and target directories for the RHSM ssl cert that we tell customers need
+# to use to access repositories managed by subscription-manager
+_RHSM_PRODUCT_CERT_SOURCE_DIR = os.path.join(utils.DATA_DIR, "rhel-certs")
+_RHSM_PRODUCT_CERT_TARGET_DIR = "/etc/pki/product-default"
+
+
+class InstallRedHatCertForYumRepositories(actions.Action):
+    id = "INSTALL_RED_HAT_CERT_FOR_YUM"
+
+    def run(self):
+        super(InstallRedHatCertForYumRepositories, self).run()
+
+        # We need to make sure the redhat-uep.pem file exists since the
+        # various Red Hat yum repositories (including the convert2rhel
+        # repo) use it.
+        # The subscription-manager-rhsm-certificates package contains this cert but for
+        # example on CentOS Linux 7 this package is missing the cert due to intentional
+        # debranding. Thus we need to ensure the cert is in place even when the pkg is installed.
+        repo_cert = cert.PEMCert(_REDHAT_CDN_CACERT_SOURCE_DIR, _REDHAT_CDN_CACERT_TARGET_DIR)
```

Comment:
```suggestion
        logger.task("Convert: Install cdn.redhat.com SSL CA certificate")
        repo_cert = cert.PEMCert(_REDHAT_CDN_CACERT_SOURCE_DIR, _REDHAT_CDN_CACERT_TARGET_DIR)
```

Without the Task-level log it'd be hard to find in the terminal output which currently looks like this:
```
BACKUP_REPOSITORY has succeeded
Certificate redhat-uep.pem copied to /etc/rhsm/ca/.
INSTALL_RED_HAT_CERT_FOR_YUM has succeeded
```